packages/flutter/lib/src/widgets/autocomplete.dart

```diff
@@ -0,0 +1,665 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/scheduler.dart';
+
+import 'basic.dart';
+import 'container.dart';
+import 'editable_text.dart';
+import 'framework.dart';
+import 'overlay.dart';
+
+/// A type for getting some list of results based on a String.
+///
+/// See also:
+///   * [AutocompleteController.getResults], which is of this type.
+typedef AutocompleteResultsGetter<T> = List<T> Function(TextEditingValue textEditingValue);
+
+/// A type for indicating the selection of an autocomplete result.
+typedef AutocompleteOnSelected<T> = void Function(T? result);
+
+/// A builder for the selectable results given the current autocomplete field
+/// text.
+typedef AutocompleteResultsBuilder<T> = Widget Function(
+  BuildContext context,
+  AutocompleteOnSelected<T> onSelected,
+  List<T> results,
+);
+
+/// A builder for the field in autocomplete.
+typedef AutocompleteFieldBuilder = Widget Function(
+  BuildContext context,
+  TextEditingController textEditingController,
+  VoidCallback onFieldSubmitted,
+);
+
+// A type for getting a String from some option.
+typedef AutocompleteOptionToString<T> = String Function(T option);
+
+// TODO(justinmc): Link to Autocomplete and AutocompleteCupertino when they are
```

Comment:
Do you need to import the relevant library in order to link to something in the docs?  Like `/// See [Autocomplete]`.  If so, then yeah we shouldn't link to them here.

Reply:
Ah thanks for looking into that.  I guess we should mention the other classes but not link to them, then.  I'll update this TODO.

Reply:
That's so interesting -- It works on the API docs, but trying to use `Go to Definition` on my IDE just gives a no definition error

Reply:
Yes, it will fail with an invalid link. And dartdoc uses a much broader set of imports to find symbols than the file itself has, just so that you can have docs links that would be layer violations if you imported the relevant code. The IDE only uses the analyzer to find symbols, so it can't find the symbols that violate layers.